### ISSUE TITLE
bug(web): when a keyboard is activated it may not be reflected in `.activeKeyboard` if not yet loaded

### ISSUE THREAD

**Issue by mcdurdin** [keymanapp/keyman#5730]
**Describe the bug**

It is possible to call `setActiveKeyboard()` for a keyboard that is registered but not yet loaded. In that situation, an event handler on `kmw.keyboardchange` that calls `keyman.getActiveKeyboard()` will find that it does not match the keyboard passed in to the event handler.

**Expected behavior**

The keyboard should be marked as active in KeymanWeb at the time that the event handler is called.

**Screenshots**

This log shows the current behaviour, logging parameter passed in to the event, result of `keyman.getActiveKeyboard()` and `keyman.core.activeKeyboard`:

```
keyboardregistered:{"internalName":"Keyboard_khmer_angkor","language":"khmer_angkor","keyboardName":"khmer_angkor","languageCode":"en"} [active=;undefined]
keyboardchange:{"internalName":"Keyboard_khmer_angkor","languageCode":"en","indirect":false} [active=;null]
keyboardregistered:{"internalName":"Keyboard_nko","language":"nko","keyboardName":"nko","languageCode":"en"} [active=;null]
keyboardloaded:{"keyboardName":"Keyboard_khmer_angkor"} [active=;null]
keyboardloaded.afterTimeout:{"keyboardName":"Keyboard_khmer_angkor"} [active=Keyboard_khmer_angkor;[object Object]]
```

Note also how even on the `keyboardloaded` event, we don't get the active keyboard set; only after control returns (a 10ms timeout) do we see the correct value.

---

**KeymanWeb:**
 - OS: any 
 - Browser: any 
 - KeymanWeb Version: 15.0

**Comment by jahorton** [comment]
I can't say for sure, but this may be related to some of the keyboard Promise API stuff.

**Comment by jahorton** [comment]
> **Expected behavior**
> 
> The keyboard should be marked as active in KeymanWeb at the time that the event handler is called.

Making sure I understand the nature of the bug:  the event should not be raised until the keyboard is both loaded and active, correct?  So if we can't immediately load it, the event should be deferred until the keyboard is ready?

Also, is there a pre-existing test page for this?  Seems like you had logging (of some sort) set up to get that report, but I'm not immediately finding a test page that matches.  And, as we do have a lot of test pages, I figured I'd best check first.

**Comment by mcdurdin** [comment]
> Making sure I understand the nature of the bug: the event should not be raised until the keyboard is both loaded and active, correct? So if we can't immediately load it, the event should be deferred until the keyboard is ready?

Yes. It makes no sense to raise the event if the keyboard is not ready.

> Also, is there a pre-existing test page for this? Seems like you had logging (of some sort) set up to get that report, but I'm not immediately finding a test page that matches.

No, not really. I discovered this in work on #5723 (which is the Developer js test page) but I think I just threw in a `console.log()` at the front of each event handler.
